theano/tensor/tests/test_opt.py

```diff
@@ -3482,6 +3482,61 @@ def test_local_fill_useless():
     f(m_, x_)
 
 
+def test_local_elemwise_sub_zeros():
+    # Test opt local_elemwise_sub_zeros
+    # We test separetly for scalars, vectors and matrices
+    scalar = T.scalar()
+    vect = T.vector()
+    mat = T.matrix()
+
+    rng = numpy.random.RandomState(seed=utt.fetch_seed())
+    scalar_val = rng.rand(1)[0]
+    vect_val = rng.rand(5)
+    mat_val = rng.rand(3, 2)
+
+    mode = theano.compile.get_default_mode()\
+            .excluding('canonicalize', 'uncanonicalize',\
+                       'ShapeOpt', 'local_fill_to_alloc',\
+                       'local_elemwise_alloc')\
+            .including('local_elemwise_sub_zeros')
+
+    # Test scalar minus scalar
+    f = function([scalar], scalar-scalar, mode=mode)
+    # Check optimized graph is correct
+    assert isinstance(f.maker.fgraph.toposort()[0].op, T.Elemwise)
```

Comment:
I'm confused, if there is still an Elemwise in the graph, doesn't it mean that the optimization has not been applied?